app/pages/login/login.html

```diff
@@ -6,23 +6,25 @@
 
 <ion-content>
 
-  <form (submit)="submitLogin(username, password)">
+  <form [ngFormModel]="loginForm" (ngSubmit)="submitLogin(loginForm.value.username, loginForm.value.password)">
 
     <ion-list>
-      <ion-item>
-        <ion-input #username type="text" placeholder="Username"></ion-input>
+      <ion-item [class.error]="!username.valid && username.touched">
+        <ion-label floating>Username</ion-label>
+        <ion-input type="text" [ngFormControl]="username"></ion-input>
```

Comment:
I think you can simply add `required` here on the input as well. Angular 2 should notice the attribute there. So the manual validators in the Component shouldn't be unnecessary?